ERCS/erc-7785.md

```diff
@@ -0,0 +1,107 @@
+
+### Chain Identifier Spoofing and Replay Attacks
+
+An important property of the centralized list is that it keeps a one-to-one correspondence between names and indentifiers.
+
+Without this property, an attacker could register a fresh name pointing to an existing identifier. For example `my-testnet` could point to mainnet `0x1`. A user could be tricked into signing a transaction for the innocent looking `my-testnet` while actually signing a transaction for mainnet, a transaction that we attacker can then replay.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Extending chain identifiers
+
+Current chain identifiers are usually chosen arbitrarily to be short. While these identifiers are convenient on a small scale, as their number increases it is more desirable to draw them from a larger space.
+
+We propose to extend the size of identifiers to 32 bytes and to derive them using a cryptographic hash function.
+The input to the function MUST contain the chain name and MAY contain additional information.
+
+An example for a L2:
+```
+chain_id = Keccak-256(CHAIN_NAME, SETTLEMENT_CHAIN_ID, VERSION, DEPLOYER_CONTRACT_ADDRESS, SALT)
+```
+where:
+- `L1_CHAIN_ID` is the id of the L1 where the L2 settles, it could be Mainnet or a testnet.
+- `VERSION` is to separate the domain of the hash function with an arbitrary string
+- `BRIDGE` is the address of the L2 on the L1
+
+### Chain name resolution
+
+Any ENS name can resolve to a chain identifier as specified in ENSIP-11. The name should resolve to a record containing not only the chain identifier, but also all the optional information necessary to verify the identifier.
+
+For example the chain name `rollup` can be converted to a chain identifier on Mainnet by resolving:
+```
+rollup.eth -> {version : uint, bridge : address, chain_id : chain_id}
+```
+and then verified using:
+```
+chain_id == hash("rollup", 0x1, version, bridge)
+```
+
+## Rationale
+
+<!--
+  The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages.
+
+  The current placeholder is acceptable for a draft.
+
+  TODO: Remove this comment before submitting
+-->
+
+TBD
+
+## Backwards Compatibility
+
+Existing identifiers, that were not derived using the scheme above, can be supported using a reverse mapping from chain identifiers to chain names, so that one can check for uniqueness.
+
+For example the chain name `legacy-rollup.eth` can be resolved to the chain identifier `0x123`.
+Then `0x123` can be resolved in the `chainid.reverse` domain to a `chain_name`.
+If `chain_name == legacy-rollup` then the mapping is valid.
```

Comment:
what if that name is already registered, tho? or what if someone registers `legacy-rollup.eth` to attack/break this because they don't like, or compete with, `legacy-rollup`?

if this goes into action, i'd recommend ENS consider blocklisting/squatting/otherwise-protecting all those chain_names from future registration to prevent abuses and hijinx. plenty of people would love to block/break this fallback for major chains that DON'T move quickly and register L2 names.

Reply:
@bumblefudge, this is addressed in the Bootstrapping and handover section (line 89). All existing rollups will be registered to avoid this scenatio